### ISSUE TITLE
Problem when moving a site

### ISSUE THREAD

**Issue by mpbt47** [simple-jwt-login/mailpoet-integration#1]
Hi,
I moved my site to a new host. I changed the URL's. MailPoet works fine, Simple JWT Logins works fine, everything works fine except that mailpoet integration continues to point to the old URL when I use the shortcode. What is the problem and what can I do ?  
Thank you.

**Comment by nicumicle** [comment]
Hi, 
I believe the issue comes from the site migration.

The site URL is received from the WordPress `site_url()` function and this is read from the database. 

For example, on my local setup, I can run a query and check the "siteurl":

Query:
```sql
SELECT * FROM wp_options WHERE option_name="siteurl";
```

Result:
| option_id | option_name | option_value        | autoload |
|-----------|-------------|---------------------|----------|
| 1         | siteurl    | http://localhost:88 | yes      |

You might have here the old value for the `option_name=siteurl`.

To fix this, you will need to update the DB with the following query:

```sql
    UPDATE wp_options SET option_value="http://newsite.com/" WHERE option_name="siteurl" LIMIT 1;
```

In this query, please replace "http://newsite.com/" with the actual website.

Please let me know if this solves your problem.

**Comment by mpbt47** [comment]
Hi,
thank you for your message
I have checked the table wp_options in the database, the value of
option_name is site2 (the new one) which is normal otherwise my new site
will not function. I would be sent back to my old site when I would
navigate within the site.
I have deleted MailPoet, including all the database tables, and reinstalled
it : no change. But I know the MailPoet writes also in other tables,
including the options table. Could the problem come from these other inputs
from MailPoet ?

**Comment by nicumicle** [comment]
Hi,

The issue is likely related to how we get the site_url. 
The URL for autologin is built here:
https://github.com/simple-jwt-login/mailpoet-integration/blob/master/mailpoet.php#L59

Do you have some caching plugins?

Also, have you checked the simple-jwt-login plugin to see if it displays the correct URL for the examples?

I suspect that, if here you get a different URL, you might have the same issue in the simple-jwt-login plugin.

**Comment by mpbt47** [comment]
Hi,

I do not use a cache plugin.
The examples of the simple-jwt-login pluginshow site2, the new one.

I have seen the following line in your program :

$url = $wordPressData->getSiteUrl() . '?rest_route=/'
            . $jwtSettings->getGeneralSettings()->getRouteNamespace() .
'autologin&JWT=' . $jwt;

The problem might come from getSiteUrl() ?
Is getSiteUrl()a wordpress function or is it a function that you have made ?

**Comment by nicumicle** [comment]
Yes, getSiteUrl() is just a wrapper over a [get_site_url()](https://developer.wordpress.org/reference/functions/get_site_url/) WordPress function.

But the same wrapper is used in the Simple-JWT-Login plugin.
This means that, somehow when the MailPoet plugin calls the custom code generator, WordPress detects a different site URL.

It would be easier to just do some debugging.  

If it's ok for you, let's please continue this conversation on contact@simplejwtlogin.com so that we can find together a solution on how we can fix this issue.

**Comment by mpbt47** [comment]
OK, let's continue this conversation on ***@***.***

Michel

**Comment by mpbt47** [comment]
Hi,

I did not receive any message from you for a week.
Of course, I would like to have my site debugged as soon as possible, but I
am not impatient, and I fully realize that you probably have many other
things to do.
I just would like to be sure that you are receiving the messages I sent to
***@***.*** and that you are not waiting for a reply from my
part.

I do appreciate your help and the attention you give to my problem.

Michel Berthelet

**Comment by nicumicle** [comment]
Hello, 

I'm sorry for not getting back to you sooner. 
I just did some checks, and I pushed a fix.

Can you please download the plugin from: https://github.com/simple-jwt-login/mailpoet-integration/archive/refs/heads/master.zip and test it on your side?

**Comment by mpbt47** [comment]
Thank you for your message. I am ready to start debugging whenever you
like; could be in a few hours, days or weeks, it is your call.

Michel Berthelet

**Comment by mpbt47** [comment]
Hi,

I am not sure, but I suspect that my emails sent to
***@***.*** are considered spam.
Could you please check, this email is sent to both addresses?
Thank you. Regards.

Michel Berthelet
